helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -162,6 +162,20 @@ public InstanceConfig getInstanceConfig(String clusterName, String instanceName)
   }
 
   @Override
+  public boolean setInstanceConfig(String clusterName, String instanceName,InstanceConfig instanceConfig) {
+    String instanceConfigPath =
+        PropertyPathConfig.getPath(PropertyType.CONFIGS, clusterName, ConfigScopeProperty.PARTICIPANT.toString(),
+            instanceName);
+    if (!_zkClient.exists(instanceConfigPath)) {
+      throw new HelixException("instance" + instanceName + " does not exist in cluster " + clusterName);
+    }
+
+    HelixDataAccessor accessor = new ZKHelixDataAccessor(clusterName, new ZkBaseDataAccessor<ZNRecord>(_zkClient));
+    Builder keyBuilder = accessor.keyBuilder();
+    return accessor.setProperty(keyBuilder.instanceConfig(instanceName), instanceConfig);
```

Comment:
this will override the existing configuration. Helix relies on HELIX_HOST and HELIX_PORT in this instanceConfig. We have two options here
-- Ensure that these two properties are set. If not, copy them from previous instanceConfig.
-- Change the method name to updateInstanceConfig and merge the old and new instanceConfigs. We can do this using oldInstanceConfig.getRecord().merge(newInstanceConfig.getRecord());